src/state/apis/swappers/swappersApi.ts

```diff
@@ -1,9 +1,88 @@
+import { selectFeatureFlags } from 'state/slices/selectors'
+import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
 
 import { BASE_RTK_CREATE_API_CONFIG } from '../const'
 
 export const swappersApi = createApi({
   ...BASE_RTK_CREATE_API_CONFIG,
   reducerPath: 'swappersApi',
-  endpoints: _build => ({}),
+  endpoints: build => ({
+    getTradeQuote: build.query<ApiQuote[], GetTradeQuoteInput>({
+      queryFn: async (getTradeQuoteInput: GetTradeQuoteInput, { getState, dispatch }) => {
+        const queryStartTime = Date.now()
+        const state = getState() as ReduxState
+        const { sendAddress, receiveAddress } = getTradeQuoteInput
+        const isCrossAccountTrade = sendAddress !== receiveAddress
+        const { LifiSwap, ThorSwap }: FeatureFlags = selectFeatureFlags(state)
+        const quotes: (Result<TradeQuote2, SwapErrorRight> & {
+          swapperName: SwapperName
+        })[] = []
+        const quoteHelperArgs = [getTradeQuoteInput, state] as const
+        if (LifiSwap)
+          try {
+            const lifiTradeQuote: Result<TradeQuote2, SwapErrorRight> =
+              await getLifiTradeQuoteHelper(...quoteHelperArgs)
+            quotes.push({
+              ...lifiTradeQuote,
+              swapperName: SwapperName.LIFI,
+            })
+          } catch (error) {
+            console.error('[getLifiTradeQuoteHelper]', error)
+          }
+        if (ThorSwap)
+          try {
+            const thorTradeQuote: Result<TradeQuote2, SwapErrorRight> =
+              await getThorTradeQuoteHelper(...quoteHelperArgs)
+            quotes.push({
+              ...thorTradeQuote,
+              swapperName: SwapperName.Thorchain,
+            })
+          } catch (error) {
+            console.error('[getThorTradeQuoteHelper]', error)
+          }
+        const quotesWithInputOutputRatios = quotes
+          .map(result => {
+            const quote = result && result.isOk() ? result.unwrap() : undefined
+            const error = result && result.isErr() ? result.unwrapErr() : undefined
+            const inputOutputRatio = quote
+              ? getInputOutputRatioFromQuote({
+                  state,
+                  quote,
+                  swapperName: result.swapperName,
+                })
+              : -Infinity
+            return { quote, error, inputOutputRatio, swapperName: result.swapperName }
+          })
+          .filter(result => {
+            const swapperSupportsCrossAccountTrade = isCrossAccountTradeSupported(
+              result.swapperName,
+            )
+            return !isCrossAccountTrade || swapperSupportsCrossAccountTrade
+          })
+        const orderedQuotes: ApiQuote[] = orderBy(
+          quotesWithInputOutputRatios,
+          ['inputOutputRatio', 'swapperName'],
+          ['asc', 'asc'],
+        )
+
+        dispatch(tradeQuoteSlice.actions.setQuotes({ quotes: orderedQuotes, queryStartTime }))
+
+        return { data: orderedQuotes }
```

Comment:
we dont want use using the result without the safety of your setter `setQuotes`

```suggestion
        return { data: null }
```

Reply:
Very true. Actioned in https://github.com/shapeshift/web/pull/4847/commits/c74339b12b9b75e9f5930866150318945d88049c.